presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -155,22 +165,37 @@ public PlanNodeDecorrelator(PlanNodeIdAllocator idAllocator, Lookup lookup)
             }
 
             DecorrelationResult childDecorrelationResult = childDecorrelationResultOptional.get();
+
             if (childDecorrelationResult.atMostSingleRow) {
                 return childDecorrelationResultOptional;
             }
 
-            if (node.getCount() != 1) {
+            if (childDecorrelationResult.getConstantSymbols().isEmpty()) {
```

Comment:
how is this possible? Did we try to decorelate  uncorrelated subquery?

Reply:
Constant symbols might be empty when correlated predicates are of form: `corr_x > uncorr_x`

Reply:
Let's keep this check specific to `rewriteLimitWithRowCountGreaterThanOne` or `rewriteLimitWithRowCountOne` as this code is already tricky